### PR TITLE
Hassan

### PR DESCRIPTION
**Pull request by hassanmohaideen**
Merging hassan into master

### CODE REVIEW

File: mat/models/medicine.cto [hunk 1]
```diff
@@ -1,136 +1,232 @@
...
+  o Business currentOwner
+  o Address destinationAddress
+  o Address sourceAddress
+  o Address[] locations
+  --> Order order
 }
 
 /**
- * Defines a contract between a Manufacturer and an Retailer to ship using
+ * Defines a contract between a Manufacturer and an Distributor to ship * using
```
Comment (Areax) [inline mat/models/medicine.cto]: any reason why there's a * here?

File: mat/models/medicine.cto [hunk 2]
```diff
@@ -1,136 +1,232 @@
...
+  o Address billingAddress
+  o Status status
+  o Business receiver
+  o Business sender
+  --> Contract[] itemContracts
 }
 
 /**
- * A shipment being tracked as an asset on the ledger
+ * shipments can have multiple orders since multiple orders may be     * going to the same
```
Comment (Areax) [inline mat/models/medicine.cto]: Any reason why there's a * here?

File: mat/models/medicine.cto [hunk 3]
```diff
@@ -1,136 +1,232 @@
...
+ //Employee - UPDATE using updateContract Transaction
+ //Employee - DELETE using deleteContract Transaction
+asset Contract identified by contractId 
+{
   o String contractId
-  --> Manufacturer manufacturer
-  --> Carrier carrier
-  --> Retailer retailer
-  o DateTime arrivalDateTime
+  o ItemType itemToBuy
```
Comment (Areax) [inline mat/models/medicine.cto]: Now contracts have orders, not items correct?

File: npm-debug.log [hunk 4]
```diff
@@ -0,0 +1,20 @@
+0 info it worked if it ends with ok
```
Comment (Areax) [inline npm-debug.log]: do we need this `debug.log`?

File: mat/models/medicine.cto (hunk 3 above)
Comment (hassanmohaideen) [reply mat/models/medicine.cto]: No more orders. I removed them from the model and access control compelete

File: mat/models/medicine.cto [hunk 5]
```diff
@@ -116,20 +115,19 @@ asset Shipment identified by shipmentId
   o Address destinationAddress
   o Address sourceAddress
   o Address[] locations
-  --> Order order
+  --> Contract contract
+  --> Item[] items
```
Comment (Areax) [inline mat/models/medicine.cto]: I still believe we should use shipments

where one shipment contains arrays of certain items...like packages?  I'm not entirely sure how the entire system works, but I'd imagine that all the items don't get shipped in all at once.

File: mat/models/medicine.cto [hunk 6]
```diff
@@ -139,7 +137,7 @@ asset Contract identified by contractId
   o Business sellingBusiness
   o Business buyingBusiness
   o Double unitPrice
-  o Double quantity
+  o Double quantity
```
Comment (Areax) [inline mat/models/medicine.cto]: I think quantity should go along with the shipment.  Put each shipment with a unitPrice maybe?

File: mat/models/medicine.cto [hunk 7]
```diff
@@ -213,6 +206,11 @@ transaction deleteContract
 {
 }
 
+//Creates a shipment
+transaction createShipment
```
Comment (Areax) [inline mat/models/medicine.cto]: Do we need a way for other parties to accept the contract?

Comment (hassanmohaideen) [reply mat/models/medicine.cto]: I'm pretty sure that would be written in the logic.js file.

File: mat/models/medicine.cto (hunk 5 above)
Comment (hassanmohaideen) [reply mat/models/medicine.cto]: That's what I changed shipments too. 
Shipments now contains an array of items to represent what is in the shipment.

File: mat/models/medicine.cto (hunk 6 above)
Comment (hassanmohaideen) [reply mat/models/medicine.cto]: We can get the quantity from the size of the array of the items in the shipment.

Comment (Areax) [reply mat/models/medicine.cto]: So this should be removed correct?  Quantity isn't needed in a contract.

File: mat/models/medicine.cto (hunk 3 above)
Comment (Areax) [reply mat/models/medicine.cto]: not entirely sure why it should be `itemToBuy` though.  What does that mean?

File: mat/models/medicine.cto (hunk 5 above)
Comment (Areax) [reply mat/models/medicine.cto]: I think that it should be backwards and that Contracts should own shipments, not the other way around

File: mat/models/medicine.cto [hunk 8]
```diff
@@ -115,7 +116,7 @@ asset Shipment identified by shipmentId
   o Address destinationAddress
   o Address sourceAddress
   o Address[] locations
-  --> Contract contract
+  --> Contract[] contract
```
Comment (Areax) [inline mat/models/medicine.cto]: Shipment has one contract

**Comment by hassanmohaideen** [comment]
Merged hassan into master

**Pull request merged**